src/openfl/_internal/renderer/opengl/GLVAORenderHelper.hx

```diff
@@ -51,16 +51,17 @@ class GLVAORenderHelper {
 			renderSession.shaderManager.updateShader (shader);
 			shader.__skipEnableVertexAttribArray = false;
 			
-			bitmapData.__vaoContext = vaoContext;
-			var hasVAO: Bool = bitmapData.__vao != null;
-			if (!hasVAO) {
+			var vaoUpToDate: Bool = true;
```

Comment:
minor nitpick: i think for better reading flow it would be better to have this inverted, something like `vaoDirty`, so later we have a nice check like `if (vaoDirty || bitmapData.isBufferDirty(...))`.